recipes/libqasm/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+        "fPIC": [True, False],
+        "build_python": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_python": False
+    }
+
+    @property
+    def _should_build_test(self):
+        return not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+    @property
+    def _min_cppstd(self):
+        return "20"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10",
+            "clang": "13",
+            "apple-clang": "14",
+            "Visual Studio": "16",
+            "msvc": "192"
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        # https://github.com/QuTech-Delft/libqasm/blob/0.6.3/src/CMakeLists.txt#L234
+        self.options["antlr4-cppruntime"].shared = self.options.shared
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("tree-gen/1.0.7")
+        self.tool_requires("zulu-openjdk/11.0.19")
+        if self.settings.arch == "wasm":
+            self.tool_requires("emsdk/3.1.50")
+        if self._should_build_test:
+            self.test_requires("gtest/1.14.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd},"
+                                            f"which your compiler does not support.")
+
+        if self.dependencies["antlr4-cppruntime"].options.shared != self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} requires antlr4-cppruntime to be built with the same shared option value.")
+
+    def requirements(self):
+        self.requires("fmt/10.2.1", transitive_headers=True)
+        self.requires("range-v3/0.12.0", transitive_headers=True)
+        self.requires("tree-gen/1.0.7", transitive_headers=True, transitive_libs=True)
+        if not self.settings.arch == "wasm":
+            self.requires("antlr4-cppruntime/4.13.1", transitive_headers=True)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.variables["LIBQASM_BUILD_PYTHON"] = self.options.build_python
+        tc.variables["LIBQASM_BUILD_TESTS"] = self._should_build_test
+        tc.generate()
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+    def build(self):
+        cmake = CMake(self)
```

Comment:
I would recommend removing `-Werror` from https://github.com/QuTech-Delft/libqasm/blob/v3-0.5.7/src/CMakeLists.txt#L296 using `replace_in_file`. It's not a good idea to include it in public builds, since it's prone to fail with new warnings and checks added with new compiler versions.

Reply:
Hi @valgur, thanks for your comment. I will think about this, but prefer leaving them on at the moment.

Reply:
I've changed my mind. I'm using `replace_in_file` to remove it.

Reply:
Hmm, you are now removing `/WX` only, but not `-Werror`. Is that intentional?

Reply:
Thanks for noticing,  @valgur! That was unwanted. I'll do it next Tuesday once I'm back to work.

Reply:
Done!